sdk/provisioning/Azure.Provisioning/tests/SampleTests.cs

```diff
@@ -0,0 +1,327 @@
+                }
+                properties: {
+                    allowBlobPublicAccess: false
+                    isHnsEnabled: true
+                }
+            }
+
+            resource blobs 'Microsoft.Storage/storageAccounts/blobServices@2023-01-01' = {
+                name: 'default'
+                parent: storage
+            }
+
+            output blobs_endpoint string = storage.properties.primaryEndpoints.blob
+            """)
+        .Lint()
+        .ValidateAndDeployAsync(
+            async deployment =>
+            {
+                // Create a client
+                BlobServiceClient storage = InstrumentClient(
+                    deployment.CreateClient(
+                        blobs!,
+                        TestEnvironment.Credential,
+                        InstrumentClientOptions(new BlobClientOptions())));
+
+                // Make sure the output matched correctly
+                Assert.AreEqual(storage.Uri.AbsoluteUri, endpoint!.Value.Value);
+
+                // Make a service call and verify it doesn't throw
+                Response<BlobServiceProperties> result = await storage.GetPropertiesAsync();
+                Assert.AreEqual(200, result.GetRawResponse().Status);
+            });
+    }
+
+    [RecordedTest]
+    public async Task SimpleContainerApp()
+    {
+        await using Trycep test = CreateBicepTest();
+        await test.Define(
+            ctx =>
+            {
+                BicepParameter location = BicepParameter.Create<string>(nameof(location), BicepFunction.GetResourceGroup().Location);
+                BicepParameter principalId = BicepParameter.Create<string>(nameof(principalId), "");
+                BicepParameter tags = BicepParameter.Create<object>(nameof(tags), new BicepDictionary<string>());
+                UserAssignedIdentity mi =
+                    new(nameof(mi))
+                    {
+                        Location = location,
+                        Tags = tags,
+                    };
+                ContainerRegistryService acr =
+                    new(nameof(acr))
+                    {
+                        Location = location,
+                        Sku = new ContainerRegistrySku() { Name = ContainerRegistrySkuName.Basic },
+                        Tags = tags,
+                        Identity =
+                            new ManagedServiceIdentity
+                            {
+                                ManagedServiceIdentityType = ManagedServiceIdentityType.SystemAssignedUserAssigned,
+                                UserAssignedIdentities =
+                                {
+                                    // TODO: Decide if we want to invest in a less janky way to use expressions as keys
+                                    { BicepFunction.Interpolate($"{mi.Id}").Compile().ToString(), new UserAssignedIdentityDetails() }
+                                }
+                            }
```

Comment:
Or maybe you're trying to jam an expression somewhere I didn't plan for an expression to fit?  You can probably still make it work.